base_classes/NXlens_em.nxdl.xml

```diff
@@ -0,0 +1,110 @@
+#
+# For further information, see http://www.nexusformat.org
+-->
+<!--
+has_part pole_piece https://purls.helmholtz-metadaten.de/emg/EMG_00000039-->
+<!--
+more consolidation to harvest from a generic component base class
+with other research fields (MPES, XPS, OPT) could be useful-->
+<definition xmlns="http://definition.nexusformat.org/nxdl/3.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" category="base" type="group" name="NXlens_em" extends="NXobject" xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd">
+    <doc>
+         Base class for an electro-magnetic lens or a compound lens.
+         
+         For :ref:`NXtransformations` the origin of the coordinate system is placed
+         in the center of the lens (its polepiece, pinhole, or another
+         point of reference). The origin should be specified in the :ref:`NXtransformations`.
+         
+         For details of electro-magnetic lenses in the literature
+         see e.g. `L. Reimer &lt;https://doi.org/10.1007/978-3-540-38967-5&gt;`_
+    </doc>
+    <!--user perspective-->
+    <field name="name" type="NX_CHAR">
+        <doc>
+            Name of the lens.
+        </doc>
+    </field>
+    <field name="description" type="NX_CHAR">
+        <doc>
+            Ideally, use instances of ``identifierNAME`` to point to a resource
+            that provides further details.
+
+            If such a resource does not exist or should not be used, use this free text,
+            although it is not recommended.
+        </doc>
+    </field>
+    <field name="power_setting" type="NX_NUMBER" units="NX_ANY">
+        <doc>
+             Descriptor for the lens excitation when the exact technical details
+             are unknown or not directly controllable as the control software of
+             the microscope does not enable or was not configured to display these
+             values (for end users).
+             
+             Although this value does not document the exact physical voltage or
+             excitation, it can still give useful context to reproduce the lens
+             setting, provided a properly working instrument and software sets the lens
+             into a similar state to the technical level possible when no more
+             information is available physically or accessible legally.
+        </doc>
+    </field>
+    <field name="mode" type="NX_CHAR">
+        <doc>
+             Descriptor for the operation mode of the lens when other details are not
+             directly controllable as the control software of the microscope
+             does not enable or is not configured to display these values.
+             
+             Like value, the mode can only be interpreted for a specific microscope
+             but can still be useful to guide users as to how to repeat the measurement.
+        </doc>
+    </field>
+    <!--control level perspective-->
+    <field name="voltage" type="NX_NUMBER" units="NX_VOLTAGE">
```

Comment:
Would it make sense to consider measured vs set points?  E.g., field `voltage` records the measured voltage while `voltage_set` describes the set point.

Reply:
Another point: the name is somewhat unfortunate -- just mirroring the units.

Would something like `name="control_voltage"` make sense?

Reply:
OK.  Thanks for the feedback.